tradeit-android-sdk/src/main/java/it/trade/android/sdk/manager/TradeItLinkedBrokerManager.java

```diff
@@ -51,6 +71,146 @@ private void loadLinkedBrokersFromSharedPreferences() throws TradeItRetrieveLink
+                public Iterable<TradeItLinkedBrokerParcelable> apply(@NonNull List<TradeItLinkedBrokerParcelable> tradeItLinkedBrokerParcelables) throws Exception {
+                    return tradeItLinkedBrokerParcelables;
+                }
+            }).map(new Function<TradeItLinkedBrokerParcelable, Single<TradeItLinkedBrokerParcelable>>() {
+                @Override
+                public Single<TradeItLinkedBrokerParcelable> apply(@NonNull final TradeItLinkedBrokerParcelable linkedBrokerParcelable) throws Exception {
+                    return Single.create(new SingleOnSubscribe<TradeItLinkedBrokerParcelable>() {
+                        @Override
+                        public void subscribe(@NonNull final SingleEmitter<TradeItLinkedBrokerParcelable> emmiter) throws Exception {
+                            linkedBrokerParcelable.authenticateIfNeeded(new TradeItCallbackWithSecurityQuestionImpl<List<TradeItLinkedBrokerAccountParcelable>>() {
+                                @Override
+                                public void onSecurityQuestion(TradeItSecurityQuestion securityQuestion) {
+                                    callback.onSecurityQuestion(securityQuestion, this);
+                                }
+
+                                @Override
+                                public void onSuccess(List<TradeItLinkedBrokerAccountParcelable> type) {
+                                    emmiter.onSuccess(linkedBrokerParcelable);
+                                }
+
+                                @Override
+                                public void onError(TradeItErrorResult error) {
+                                    Log.e(TAG, error.toString());
+                                    emmiter.onError(new RuntimeException(error.toString()));
+                                }
+
+                                @Override
+                                public void cancelSecurityQuestion() {
+                                    emmiter.onSuccess(linkedBrokerParcelable);
+                                }
+                            });
+                        }
+                    });
+                }
+            }).flatMap(new Function<Single<TradeItLinkedBrokerParcelable>, ObservableSource<TradeItLinkedBrokerParcelable>>() {
+                @Override
+                public ObservableSource<TradeItLinkedBrokerParcelable> apply(@NonNull Single<TradeItLinkedBrokerParcelable> tradeItLinkedBrokerParcelableSingle) throws Exception {
+                    return tradeItLinkedBrokerParcelableSingle.toObservable();
+                }
+            }, true).subscribe(new DisposableObserver() {
+                @Override
+                public void onNext(@NonNull Object linkedBrokerParcelable) {
+                    Log.d(TAG, "onNext: " + linkedBrokerParcelable);
+                }
+
+                @Override
+                public void onError(@NonNull Throwable e) {
+                    callback.onFinished();
```

Comment:
This seems like this will call `callback.onFinished` the first time there is any error with a linked broker, and then will not process the rest of the linked brokers. Shouldn't we just log here instead of calling the callback?